### PR TITLE
`<chrono>` parse subseconds when the underlying type supports it

### LINKED ISSUE #1952: <chrono>: parse ignores subseconds when the underlying type supports it
**Describe the bug**
With 16.10.0 and its new chrono parsers, the parse and from_stream functions do not parse fractional seconds in some cases. This is in contrast to the new `std::format` which outputs fractional seconds, creating a discrepancy between the two.

**Command-line test case**
```
C:\Test>type repro.cpp
#include <chrono>
#include <iostream>
#include <sstream>

int main() {
    auto now = std::chrono::utc_clock::now();
    auto str = std::format("{:%FT%TZ}", now);
    std::cout << now.time_since_epoch().count() << " -> " << str << " -> ";
    std::istringstream istr(str);
    std::chrono::utc_clock::time_point now_parsed;
    istr >> std::chrono::parse(std::string("%FT%TZ"), now_parsed);
    // istr.fail() == true
    std::cout << now_parsed.time_since_epoch().count() << "\n";
}
C:\Test>cl /std:c++latest /EHsc /W4 /WX repro.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30037 for x86
Copyright (C) Microsoft Corporation.  All rights reserved.

/std:c++latest is provided as a preview of language features from the latest C++
working draft, and we're eager to hear about bugs and suggestions for improvements.
However, note that these features are provided as-is without support, and subject
to changes or removal as the working draft evolves. See
https://go.microsoft.com/fwlink/?linkid=2045807 for details.

repro.cpp
Microsoft (R) Incremental Linker Version 14.29.30037.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:repro.exe
repro.obj

C:\Test>repro.exe
16226562926960288 -> 2021-06-02T17:51:05.6960288Z -> 0
```

**Expected behavior**
`now_parsed == now`, and `istr.fail() == false`

**STL version**
```
Microsoft Visual Studio Community 2019
Version 16.10.0
```

**Additional context**
From investigating further, `std::chrono::parse` can read `std::format`'s output to an extent, but refuses to read any subsecond precision, meaning it isn't able to reach the "Z" literal, causing it to fail. Trying it with a duration instead of a time_point gets a bit farther, but refuses to convert the parsed internal `_Time_parse_fields` to a duration, failing in `_Apply_duration_fields` at `_Test_bits`. For clarification, the formatting I'm using with `%FT%TZ` is to use ISO 8601.

### PR DESCRIPTION
**Pull request by MattStephanson**
- Fixes #1952.
 - Also fixes incorrect behavior when local_time is parsed with a
   UTC offset.
 - More test coverage for parsed UTC offset.

### CODE REVIEW

File: tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp [hunk 1]
```diff
@@ -888,22 +892,68 @@ void insert_leap_second(const sys_days& date, const seconds& value) {
         tzdb{my_tzdb.version, move(zones), move(links), move(leap_vec), my_tzdb._All_ls_positive && (value == 1s)});
 }
 
+void test_gh_1952() {
+    auto time_str{"2021-06-02T17:51:05.696028Z"};
+    auto fmt{"%FT%TZ"};
+    auto utc_ref = clock_cast<utc_clock>(sys_days{2021y / June / 2d} + 17h + 51min + 05s + 696028us);
```
Comment (StephanTLavavej) [inline tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp:898]: These could be top-level `const`. We're not super-rigorous about this, especially for short/simple functions in test code, but in longer/complicated test functions (this one is moderately so), it helps to indicate that we won't be mutating these variables, avoids accidental mutation, and (when done consistently) the lack of `const` on other variables indicates that we *will* be mutating them.

Comment (StephanTLavavej) [inline tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp:898]: I think we should avoid the leading zero in `05s` - although it mirrors `time_str`, it's an octal UDL, which is weird (and doesn't appear in other test code).

**Review by StephanTLavavej — COMMENTED** [review]
Looks good, thanks! I'll validate and push very minor test changes.

**Review by StephanTLavavej — APPROVED** [review]

File: tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp (hunk 1 above)
Comment (MattStephanson) [reply tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp]: An octal literal, good catch.

**Review by mnatsuhara — APPROVED** [review]
Looks good!  Thanks so much for implementing this fix :)

**Comment by StephanTLavavej** [comment]
Thanks for fixing this bug in a split second! :zap: :beetle: :smile_cat:

**Pull request merged**